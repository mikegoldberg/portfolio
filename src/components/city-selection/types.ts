export interface CitySelectionProps {
  city: string;
  onSubmit: (name: string) => void;
}
