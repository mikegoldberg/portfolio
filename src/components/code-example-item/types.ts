import { MouseEventHandler } from "react";

export interface CodeExampleItemProps {
  title: string;
  poster: string;
  onClick: MouseEventHandler;
  source: string;
}
