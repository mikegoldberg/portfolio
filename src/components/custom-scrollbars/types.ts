import { ReactElement } from "react";

export interface CustomScrollbarsProps {
  onScrollFrame: (position: number) => void;
  children?: ReactElement;
}
