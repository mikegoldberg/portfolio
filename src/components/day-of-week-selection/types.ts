type OnDaySelected = {
  (date: string): void;
};

export interface DayOfWeekSelectionProps {
  dates: Array<string>;
  activeDayOfWeek: string;
  onDaySelected: OnDaySelected;
}
