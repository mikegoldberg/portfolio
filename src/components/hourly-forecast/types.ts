export type HourlyForecastItemProps = {
  startTime: string;
  temperature: string;
  temperatureUnit: string;
  windDirection: string;
  temperatureChange: number;
  windSpeedValue: string;
  windSpeedUnit: string;
  iconLarge: string;
  shortForecast: string;
};

export interface HourlyForecastProps {
  items: Array<HourlyForecastItemProps> | unknown[];
  onScrollFrame: (position: number) => void;
}

export interface HourlyForecastHandler {
  scrollToDate: (date: string) => void;
  getDayOfWeekAtItemIdx: (idx: number) => string;
}

export interface TemperatureProps {
  temp: string;
  unit: string;
}

export interface WindDirectionProps {
  windDirection: string;
  windSpeed: string;
  windSpeedUnit: string;
}
