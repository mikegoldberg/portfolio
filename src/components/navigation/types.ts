export interface MobileHeadingProps {
  onOpen: () => void;
}

export interface NavMenuProps {
  onNavClicked?: () => void;
}

export interface NavLinkProps {
  label: string;
  to: string;
  onNavClicked?: () => void;
}
