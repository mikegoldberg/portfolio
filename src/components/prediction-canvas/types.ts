export type PredictionCanvasProps = {
  hide: boolean;
  width: number;
  height: number;
};
