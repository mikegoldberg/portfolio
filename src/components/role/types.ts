type Title = string;
type Company = string;
type Description = string;
type Skillset = Array<string>;
type Achievements = Array<string>;
type Start = string;
type End = string | null;
type Contract = boolean;

export interface RoleProps {
  title: Title;
  company: Company;
  description: Description;
  skillset: Skillset;
  achievements: Achievements;
  start: Start;
  end: End;
  contract: Contract;
}
