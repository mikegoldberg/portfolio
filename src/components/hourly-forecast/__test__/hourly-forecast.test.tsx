import { render } from "../../../test-utils";
import HourlyForecast from "..";
import { HourlyForecastProps } from "../types";

const defaultProps: HourlyForecastProps = {
  items: [],
  onScrollFrame: () => {},
};

describe("HourlyForecast", () => {
  it("renders correctly", () => {
    const { container } = render(<HourlyForecast {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
