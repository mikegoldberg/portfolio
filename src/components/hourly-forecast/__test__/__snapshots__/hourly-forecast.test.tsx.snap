// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourlyForecast renders correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-1om7uec"
    >
      <div
        class="css-0"
        data-testid="custom-scrollbars-content"
      >
        <div
          class="css-q2pi1i"
        />
      </div>
    </div>
    <div
      class="css-fxumzl"
    >
      <div
        class="chakra-slider css-1t1xvy7"
        style="position: relative; user-select: none; outline: 0; padding-top: 0px; padding-bottom: 0px;"
        tabindex="-1"
      >
        <div
          class="chakra-slider__track css-wdvtf3"
          id="slider-track-:r0:"
          style="position: absolute; top: 50%; transform: translateY(-50%); width: 100%;"
        />
        <div
          aria-label="slider-ex-4"
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="chakra-slider__thumb css-q2lm97"
          id="slider-thumb-:r0:"
          role="slider"
          style="position: absolute; user-select: none; left: calc(0% - 0px);"
          tabindex="0"
        >
          <svg
            class="css-f7nfc1"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M7 18h2V6H7v12zm4 4h2V2h-2v20zm-8-8h2v-4H3v4zm12 4h2V6h-2v12zm4-8v4h2v-4h-2z"
            />
          </svg>
        </div>
        <input
          type="hidden"
          value="0"
        />
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
